apps/admin-page/src/app/pages/dashboard/dashboard.component.ts

```diff
@@ -1,10 +1,454 @@
+    const startDate = new Date(Date.now() - dayStart * 24 * 60 * 60 * 1000);
+    const endDate = new Date(Date.now() - dayEnd * 24 * 60 * 60 * 1000);
+
+    const periodNewsletters = this.fullNewsletterList.filter(
+      (newsletter) => new Date(newsletter.createdAt) >= endDate && new Date(newsletter.createdAt) < startDate
+    );
+    return periodNewsletters;
+  }
+
+  getPercentDiff(thisPeriod: number, lastPeriod: number) {
+    return (((thisPeriod - lastPeriod) / thisPeriod) * 100).toFixed(1);
+  }
+
+  subtractDimensionSize(itemDimension: DesignDimensionEnum): number {
+    if (itemDimension === DesignDimensionEnum.small) {
+      return 135;
+    } else if (itemDimension === DesignDimensionEnum.medium) {
+      return 161;
+    } else if (itemDimension === DesignDimensionEnum.large) {
+      return 205;
+    } else {
+      return 115;
+    }
+  }
+
+  subtractPlantedTrees(order: IOrder): number {
+    if (order.plantedTrees > 1) {
+      return (order.plantedTrees - 1) * 10;
+    } else {
+      return 0;
+    }
+  }
+
+  subtractShippingCost(shippingMethod: ShippingMethodEnum, orderTotal: number): number {
+    if (shippingMethod === ShippingMethodEnum.homeDelivery && orderTotal > 350) {
+      return 25;
+    } else if (shippingMethod === ShippingMethodEnum.homeDelivery && orderTotal <= 350) {
+      return 65;
+    } else if (shippingMethod === ShippingMethodEnum.pickUpPoint && orderTotal <= 350) {
+      return 45;
+    } else {
+      return 0;
+    }
+  }
+
+  getPeriodOrderCount(period: number) {
+    const thisPeriodOrderCount = this.getPeriodOrders(period).length;
+    if (period === 14) {
+      return (this.twoWeekOrders = thisPeriodOrderCount);
+    } else if (period === 30) {
+      return (this.monthOrders = thisPeriodOrderCount);
+    } else if (period === 90) {
+      return (this.threeMonthOrders = thisPeriodOrderCount);
+    } else {
+      return (this.sixMonthOrders = thisPeriodOrderCount);
+    }
+  }
+
+  calculatePeriodOrderCountDifference(periodStart: number, periodEnd: number) {
+    const thisPeriodOrderCount = this.getPeriodOrders(periodStart).length;
+    const lastPeriodOrderCount = this.getPreviousPeriodOrders(periodStart, periodEnd).length;
+    if (periodStart === 14 && periodEnd === 28) {
+      this.twoPastWeekOrders = lastPeriodOrderCount;
+    } else if (periodStart === 30 && periodEnd === 60) {
+      this.monthPastOrders = lastPeriodOrderCount;
+    } else if (periodStart === 90 && periodEnd === 180) {
+      this.threePastMonthOrders = lastPeriodOrderCount;
+    } else {
+      this.sixPastMonthOrders = lastPeriodOrderCount;
+    }
+    return this.getPercentDiff(thisPeriodOrderCount, lastPeriodOrderCount);
+  }
+
+  getPeriodRevenue(period: number) {
+    const thisPeriodOrders = this.getPeriodOrders(period);
+    let thisPeriodRevenue = 0;
+    if (period === 14) {
+      thisPeriodOrders.forEach((order) => {
```

Comment:
you can move the thisPeriodOrder.forEach function outside of the if since this will happen in every case. Then the only thing you do inside the if statement is set what period the totalRevenue is calculated for.

Also, since its only one variable that you are using in every case of the if else's, this should be made into a switch.